src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -583,6 +594,13 @@ public boolean configure(final StaplerRequest req, final JSONObject formData)
         this.setTagNode(false);
       }
 
+      // Grab jobRegex and coerse to a boolean
+      if ( formData.getString("jobRegex").equals("true") ) {
```

Comment:
in case there is a null, you should do `"true".equals(formData.getString("jobRegex"))`

Reply:
what about `this.setJobRegex(formData.getString("jobRegex").equals("true"))`?